sympy/concrete/expr_with_limits.py

```diff
@@ -25,12 +26,12 @@ def _process_limits(*symbols):
     limits = []
     orientation = 1
     for V in symbols:
-        if isinstance(V, Symbol):
+        if isinstance(V, (Symbol, Idx)):
```

Comment:
If _Idx_ already contains the info about its range, then there should be no need to input in a second time.

Maybe also some range inconsistency tests? E.g.:
```
>>> i = Idx('i', range=(1, 4))
>>> A = IndexedBase('A')
>>> Sum(A[i], (i, 1, 7))
Exception: incompatible ranges.
```